pkg/uploader/chart_uploader.go

```diff
@@ -49,6 +50,12 @@ func (c *ChartUploader) UploadTo(ref, remote string) error {
 		return fmt.Errorf("scheme prefix missing from remote (e.g. \"%s://\")", registry.OCIScheme)
 	}
 
+	// See: https://github.com/helm/helm/issues/12728
+	path := strings.SplitN(u.Path, ":", 2)
+	if len(path) > 1 {
+		return fmt.Errorf("Version tag \"%s\" need not be passed for remote", path[1])
+	} 
```

Comment:
This check should be placed inside _ocipusher.go_. `UploadTo` has basic error handling for the URI to make sure it's valid and we have the ability to route it to a pusher. This logic is specific to OCI style things so it should go in that pusher.